week3/homework/Nodemailer/app.js

```diff
@@ -0,0 +1,76 @@
+const express = require('express');
+const exphbs = require('express-handlebars');
+const path = require('path');
+const nodemailer = require('nodemailer');
+
+const app = express();
+app.use(express.urlencoded({ extended: true }));
+
+app.engine('handlebars', exphbs());
+app.set('view engine', 'handlebars');
+
+app.use('/public', express.static(path.join(__dirname, 'public')));
+
+app.get('/', (req, res) => {
+  res.render('contact');
+})
+
+app.post('/send', (req, res) => {
+  const output = `
+  <p>You have a new contact request</p>
+  <h3>Contact Details</h3>
+  <ul>  
+    <li>Name: ${req.body.name}</li>
+    <li>Company: ${req.body.company}</li>
+    <li>Email: ${req.body.email}</li>
+    <li>Phone: ${req.body.phone}</li>
+  </ul>
+  <h3>Message</h3>
+  <p>${req.body.message}</p>
+`;
+
+
+
+  // async..await is not allowed in global scope, must use a wrapper
+  async function main() {
+
+
+    // create reusable transporter object using the default SMTP transport
+    let transporter = nodemailer.createTransport({
+      host: "",
+      port: 587,
+      secure: false, // true for 465, false for other ports
+      auth: {
+        user: '', // generated ethereal user
+        pass: '', // generated ethereal password
+      },
+      tls: {
+        rejectUnauthorized: false
+      }
+    });
+
+    // send mail with defined transport object
+    let info = await transporter.sendMail({
+      from: '"" <>', // sender address
```

Comment:
YOu leave the email info empty?

Reply:
Yes you are right, because GItHub sent me a massage that It has security problem, I used real data from one of my server than was my customer, so I removed all commits and create another commit without sensitive data.
But I test it with real data, it worked...
Thanks

Reply:
ah yeah makes sense. Generally the content come from some variable or database. I was just curious